src/dataframe/dataframe.jl

```diff
@@ -1118,32 +1141,33 @@ function Base.push!(df::DataFrame, row::Union{AbstractDict, NamedTuple}; columns
     if !(columns in (:equal, :intersect))
         throw(ArgumentError("`columns` keyword argument must be `:equal` or `:intersect`"))
     end
-    if ncol(df) == 0 && row isa NamedTuple
+    nrows, ncols = size(df)
+    targetrows = nrows + 1
```

Comment:
Just use `nrows + 1` below instead of defining a variable? Same in the next function.

Reply:
fixed